internal/builders/go/pkg/build.go

```diff
@@ -364,17 +378,54 @@ func (b *GoBuild) generateLdflags() (string, error) {
 				return "", fmt.Errorf("%w: %s", errorEnvVariableNameEmpty, name)
 			}
 			res = fmt.Sprintf("%s%s%s", res, v[:start], val)
-			found = true
 			v = v[start+len(ss)+end+len(es):]
 		}
-		if !found {
-			res = v
-		}
+
 		a = append(a, res)
 	}
+
+	// Special variables.
+	for i, v := range a {
```

Comment:
is there any way to factor out a `ReplaceVariable` function with this loop and use it for the above `Env.` vars too?

Reply:
a little tricky. I'me even thinking or removing this part of the code and just do a `ReplaceAll({{ .Value }})` instead of trying to accommodate spaces. Wdut?

Reply:
I'm going to remove this code and move to `ReplaceAll()`

Reply:
done, PTAL. We can add support for spaces later via the regex, if we want. For now it's simple.